scala/scala-impl/src/org/jetbrains/plugins/scala/lang/formatting/scalafmt/ScalafmtDynamicConfigManager.scala

```diff
@@ -224,10 +224,35 @@ object ScalafmtDynamicConfigManager {
   def isIncludedInProject(file: PsiFile): Boolean =
     instanceIn(file.getProject).isIncludedInProject(file)
 
-  def readVersion(configFile: VirtualFile): Either[Throwable, Option[String]] = {
+  private def parseHoconFile(project: Project, configFile: VirtualFile): Config = {
+    val fileDocumentManager = FileDocumentManager.getInstance
+    val emptyObj = ConfigValueFactory.fromMap(Map.empty[String, Any].asJava)
+    lazy val parseOpts: ConfigParseOptions = ConfigParseOptions.defaults
+      .setIncluder(new ConfigIncluder {
+        var fallbackOpt: Option[ConfigIncluder] = None
+        override def withFallback(fallback: ConfigIncluder): ConfigIncluder = {
```

Comment:
Can we rewrite it to conform `withIncluder` contact defined in comment?
```java
/**
   * Returns a new includer that falls back to the given includer. This is how
   * you can obtain the default includer; it will be provided as a fallback.
   * It's up to your includer to chain to it if you want to. You might want to
   * merge any files found by the fallback includer with any objects you load
   * yourself.
   *
   * It's important to handle the case where you already have the fallback
   * with a "return this", i.e. this method should not create a new object if
   * the fallback is the same one you already have. The same fallback may be
   * added repeatedly.
   *
   * @param fallback
   * @return a new includer
   */
  ConfigIncluder withFallback(ConfigIncluder fallback);
```

> Returns a new includer

(+doing this we could avoid mutable variable )

> It's important to handle the case where you already have the fallback with a "return this"